### PR TITLE
Support cloning CSR tensors

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #68939
* #68938
* #68883
* #68919
* __->__ #68882
* #68867

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk

### CODE REVIEW

File: aten/src/ATen/native/sparse/SparseCsrTensor.cpp [hunk 1]
```diff
@@ -322,5 +322,24 @@ const SparseCsrTensor& resize_as_sparse_csr_(
...
+SparseCsrTensor clone_sparse_csr(
+    const SparseCsrTensor& self,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
+  TORCH_CHECK(
+      !optional_memory_format.has_value(),
+      "unsupported memory format option ",
+      optional_memory_format.value());
+  TensorOptions options = self.options();
+  return at::native::_sparse_csr_tensor_unsafe(
+                                               self.crow_indices().clone(optional_memory_format),
```
Comment (IvanYashchuk) [inline aten/src/ATen/native/sparse/SparseCsrTensor.cpp:334]: If `optional_memory_format` has a value defined then an error is thrown above with `TORCH_CHECK` so no point to pass it here.

Comment (pearu) [reply aten/src/ATen/native/sparse/SparseCsrTensor.cpp:334]: Fixed.

**Review by IvanYashchuk — APPROVED** [review]

**Comment by pearu** [comment]
Replaced by https://github.com/pytorch/pytorch/pull/70581

**Comment by pearu** [comment]
https://github.com/pytorch/pytorch/pull/70581 landed

**Pull request closed without merging**